plugins/draco/io/DracoWriter.cpp

```diff
@@ -0,0 +1,455 @@
+    draco::GeometryAttribute ga;
+    ga.Init(t, nullptr, n, dataType, false, draco::DataTypeLength(dataType) * n, 0);
+    m_attMap[t] = m_pc->AddAttribute(ga, true, m_pc->num_points());
+}
+
+void DracoWriter::initPointCloud(point_count_t size)
+{
+    //begin initialization of the point cloud with the point count
+    m_pc->set_num_points(size);
+    //go through known draco attributes and add to the pointcloud
+    for (auto &dim: m_dims)
+    {
+        addAttribute(dim.first, dim.second);
+    }
+    for (auto &dim: m_genericDims)
+    {
+        addGeneric(dim, 1);
+    }
+}
+
+void DracoWriter::addPoint(int attId, Dimension::IdList idList, PointRef &point, PointId idx)
+{
+    //get first id in list. All ids in any given list should be the same datatype
+    Dimension::Id dim = idList[0];
+    Dimension::Type type;
+
+    if (m_userDimMap.find(dim) != m_userDimMap.end()) {
+        std::string typeStr = m_userDimMap.at(dim);
+        type = Dimension::type(typeStr);
+    } else {
+        type = Dimension::defaultType(dim);
+    }
+
+    //call addToPointCloud with appropriate datatype
+    switch (type) {
+        case Dimension::Type::Signed8:
+            addToPointCloud<int8_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Unsigned8:
+            addToPointCloud<uint8_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Signed16:
+            addToPointCloud<int16_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Unsigned16:
+            addToPointCloud<uint16_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Signed32:
+            addToPointCloud<int32_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Unsigned32:
+            addToPointCloud<uint32_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Signed64:
+            addToPointCloud<int64_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Unsigned64:
+            addToPointCloud<uint64_t>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Float:
+            addToPointCloud<float>(attId, idList, point, idx);
+            break;
+        case Dimension::Type::Double:
+            addToPointCloud<double>(attId, idList, point, idx);
+            break;
+        default:
+            throw pdal_error("Invalid type for pdal dimension " + Dimension::name(dim));
+            break;
+    }
+
+}
+
+void DracoWriter::write(const PointViewPtr view)
+{
+    //initialize pointcloud builder
+    initPointCloud(view->size());
+    //we don't know how many texture dimensions are available until here
+    parseDimensions();
+
+    PointRef point(*view, 0);
+    for (PointId idx = 0; idx < view->size(); ++idx)
+    {
+        point.setPointId(idx);
+        const auto pointId = draco::PointIndex(idx);
```

Comment:
Windows complains:

```
2021-01-20T22:12:34.1172288Z ..\plugins\draco\io\DracoWriter.cpp(365): warning C4244: 'argument': conversion from 'pdal::PointId' to 'ValueTypeT', possible loss of data
2021-01-20T22:12:34.1173277Z         with
2021-01-20T22:12:34.1173545Z         [
2021-01-20T22:12:34.1174092Z             ValueTypeT=uint32_t
2021-01-20T22:12:34.1174387Z         ]
2021-01-20T22:12:34.1174912Z D:\a\PDAL\PDAL\plugins\draco\io\DracoWriter.hpp(114): warning C4244: 'argument': conversion from 'pdal::PointId' to 'ValueTypeT', possible loss of data
2021-01-20T22:12:34.1175366Z         with
2021-01-20T22:12:34.1175677Z         [
2021-01-20T22:12:34.1176082Z             ValueTypeT=uint32_t
2021-01-20T22:12:34.1176692Z         ]
2021-01-20T22:12:34.1177323Z ..\plugins\draco\io\DracoWriter.cpp(318): note: see reference to function template instantiation 'void pdal::DracoWriter::addToPointCloud<int8_t>(int,pdal::Dimension::IdList,pdal::PointRef &,pdal::PointId)' being compiled
```

Reply:
Unsure how I can avoid these. PointId is a uint64_t and ValueTypeT is a uint32_t. They chose to limit themselves there.